cmd/nebraska/metrics.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	defaultMetricsUpdateInterval = time.Second * time.Duration(5)
+)
+
+var (
+	appInstancePerChannelGaugeMetric = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "nebraska",
+			Subsystem: "application",
+			Name:      "instances_per_channel",
+			Help:      "Number of applications from specific channel running on instances",
+		},
+		[]string{
+			"application",
+			"version",
+			"channel",
+		},
+	)
+
+	failedUpdatesGaugeMetric = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "nebraska",
+			Name:      "failed_updates",
+			Help:      "Number of failed updates of an application",
+		},
+		[]string{
+			"application",
+		},
+	)
+)
+
+// registerNebraskaMetrics registers the application metrics collector with the DefaultRegistrer.
+func registerNebraskaMetrics() error {
+	err := prometheus.Register(appInstancePerChannelGaugeMetric)
+	if err != nil {
+		return err
+	}
+	err = prometheus.Register(failedUpdatesGaugeMetric)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// getMetricsRefreshInterval returns the metrics update Interval key is set in the environment as time.Duration,
+// If not returns the default update interval.
+func getMetricsRefreshInterval() time.Duration {
+	refreshInterval, err := strconv.Atoi(os.Getenv("NEBRASKA_METRICS_UPDATE_INTERVAL"))
+	if err != nil {
+		return defaultMetricsUpdateInterval
+	}
+
+	return time.Second * time.Duration(refreshInterval)
+}
+
+// registerAndInstrumentMetrics registers the application metrics and instruments them in configurable intervals.
+func registerAndInstrumentMetrics(ctl *controller) error {
+	// register application metrics
+	err := registerNebraskaMetrics()
+	if err != nil {
+		return err
+	}
+
+	refreshInterval := getMetricsRefreshInterval()
+
+	metricsTicker := time.NewTicker(refreshInterval)
+
+	go func() {
+		for {
+			<-metricsTicker.C
+			err := calculateMetrics(ctl)
+			if err != nil {
+				logger.Error("Error updating the metrics")
+			}
+		}
+	}()
+
+	return nil
+}
+
+// calculateMetrics calculates the application metrics and updates the respective metric.
+func calculateMetrics(ctl *controller) error {
+	aipcMetrics, err := ctl.api.GetAppInstancesPerChannelMetrics()
+	if err != nil {
+		logger.Error("getMetrics - getting app instances per channel metrics", "error", err.Error())
+		return err
```

Comment:
I'd just return a descriptive error only - it's going to be logged by the caller of `calculateMetrics`.

```suggestion
		return fmt.Errorf("failed to get app instances per channel metrics: %w", err)
```